modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalSpec.java

```diff
@@ -102,11 +99,6 @@ public RankEvalSpec(StreamInput in) throws IOException {
             this.templates.put(key, value);
         }
         maxConcurrentSearches = in.readVInt();
-        int indicesSize = in.readInt();
-        indices = new ArrayList<>(indicesSize);
-        for (int i = 0; i < indicesSize; i++) {
-            this.indices.add(in.readString());
-        }
```

Comment:
This should me handled by out.writeStringArray(indices) in the request now, and shouldn't change the wire protocol so no version-dependent logic needed here I think.

Reply:
Good point, I will add a version check here.